test/domain.test.ts

```diff
@@ -839,5 +839,71 @@ describe('store: [adapter=DOMAIN]', () => {
         globalStore.initialize(adapter)(callback);
       });
     });
+
+    it('should return the response from async callback function.', async () => {
+      const callback = async () => {
+        globalStore.set({ foo: 'foo' });
+
+        functionAccessingStore();
+        const response = await asyncTask();
+
+        return response;
+      };
+
+      const functionAccessingStore = () => {
+        expect(globalStore.get('foo')).to.equal('foo');
+      };
+
+      const asyncTask = () => {
+        return new Promise((resolve) => {
+          setTimeout(() => {
+            resolve(globalStore.get('foo'));
+          }, 1);
+        });
+      };
+
+      const response = await globalStore.initialize(adapter)(callback);
+      expect(response).to.equal('foo');
+    });
```

Comment:
@HiteshShrestha Can we add another test case that will return the response from non-async callback function?